feat: add C ndarray interface and refactor implementation for `stats/base/svarianceyc`

Vinit-Pandit opened the pull request:
Resolves None

## Description

> What is the purpose of this pull request?

This pull request:

-   add C ndarray interface and refactor implementation for `stats/base/svarianceyc`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves None

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

Vinit-Pandit committed:
feat: add C ndarray interface and refactor implementation for stats/base/svarianceyc

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: passed
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

Vinit-Pandit committed:
fix: fixing cppcheck error

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Vinit-Pandit commented:
/stdlib update-copyright-years

Vinit-Pandit committed:
fix: fixing mistakes

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Vinit-Pandit committed:
fix: fixing comments

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Vinit-Pandit commented on lib/node_modules/@stdlib/stats/base/svarianceyc/docs/repl.txt:
```diff
@@ -48,20 +48,17 @@
     ~4.3333
 
-{{alias}}.ndarray( N, correction, x, stride, offset )
+
```
```suggestion
```

Vinit-Pandit committed:
fix: fixing Readme

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Vinit-Pandit committed:
style: remove extra space

Signed-off-by: Vinit Pandit <106718914+MeastroZI@users.noreply.github.com>

Vinit-Pandit committed:
Revert previous commit

Signed-off-by: Vinit Pandit <106718914+MeastroZI@users.noreply.github.com>

Vinit-Pandit committed:
fix: readme fix

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Vinit-Pandit commented on lib/node_modules/@stdlib/stats/base/svarianceyc/docs/repl.txt:
```diff
@@ -48,20 +48,17 @@
 
     // Using `N` and `stride` parameters:
```
```suggestion
    // Using `N` and stride parameters:
```

Vinit-Pandit commented on lib/node_modules/@stdlib/stats/base/svarianceyc/lib/index.js:
```diff
@@ -28,20 +28,17 @@
-* var N = floor( x.length / 2 );
 *
-* var v = svarianceyc.ndarray( N, 1, x, 2, 1 );
+* var v = svarianceyc.ndarray( x.length, 1, x, 2, 1 );
```
```suggestion
* var v = svarianceyc.ndarray( 4, 1, x, 2, 1 );
```

Vinit-Pandit committed:
style: fix spaces

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Vinit-Pandit committed:
Some fixes

Signed-off-by: Vinit Pandit <106718914+MeastroZI@users.noreply.github.com>

Vinit-Pandit committed:
style: fixing space in manifest

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svarianceyc/benchmark/benchmark.js:
```diff
@@ -40,12 +46,8 @@ var svarianceyc = require( './../lib/svarianceyc.js' );
-	for ( i = 0; i < x.length; i++ ) {
-		x[ i ] = ( randu()*20.0 ) - 10.0;
-	}
+	x = uniform( len, -10, 10, options );
```
```suggestion
	var x = uniform( len, -10, 10, options );
```

aayush0325 replied:
same comment for all other JavaScript benchmark files

aayush0325 replied:
it is better to declare and assign at the same time when the calculation is **not complex** i.e.
```javascript
var x = someFunction(); // this is better
```
we prefer this over
```javascript
var x;

x = someFunction(); // this is also okay but not preferred
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svarianceyc/examples/index.js:
```diff
@@ -18,18 +18,14 @@
-for ( i = 0; i < x.length; i++ ) {
-	x[ i ] = round( (randu()*100.0) - 50.0 );
-}
+x = discreteUniform( 10, -50, 50, {
+	'dtype': 'float32'
+});
```
```suggestion
var x = discreteUniform( 10, -50, 50, {
	'dtype': 'float32'
});
```
kindly make this change in the `README.md` examples too if applicable

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svarianceyc/include/stdlib/stats/base/svarianceyc.h:
```diff
@@ -31,7 +31,12 @@ extern "C" {
+float API_SUFFIX(stdlib_strided_svarianceyc)( const CBLAS_INT N, const float correction, const float *X, const CBLAS_INT strideX );
+
+/**
+* Computes the variance of a single-precision floating-point strided array using a one-pass algorithm proposed by Youngs and Cramer, using alternative indexing semantics.
```
```suggestion
* Computes the variance of a single-precision floating-point strided array using a one-pass algorithm proposed by Youngs and Cramer and alternative indexing semantics.
```

this is the convention that we're following across the `stats/base` namespace for now

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svarianceyc/src/main.c:
```diff
@@ -33,41 +34,53 @@
-	ix += stride;
+
+/**
+* Computes the variance of a single-precision floating-point strided array using a one-pass algorithm proposed by Youngs and Cramer, using alternative indexing semantics.
```
```suggestion
* Computes the variance of a single-precision floating-point strided array using a one-pass algorithm proposed by Youngs and Cramer and alternative indexing semantics.
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svarianceyc/src/main.c:
```diff
@@ -33,41 +34,53 @@
+    }
+    ix = offsetX;
+    sum = X[ ix ];
+	ix += strideX;
 	S = 0.0f;
 	for ( i = 2; i <= N; i++ ) {
 		di = (double)i;
 		v = X[ ix ];
 		sum += v;
 		d = (float)(di*(double)v) - sum;
 		S += (float)(1.0/(di*(di-1.0))) * d * d;
```
indentation is off

aayush0325 reviewed (changes requested):
left a round of comments

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svarianceyc/src/main.c:
```diff
@@ -33,41 +34,53 @@
+
+    n = (double)N - (double)correction;
+    if ( N <= 0 || n <= 0.0f ) {
+        return 0.0f / 0.0f; // Nan
```
```suggestion
        return 0.0f / 0.0f; // NaN
```

MeastroZI commented on lib/node_modules/@stdlib/stats/base/svarianceyc/src/main.c:
```diff
@@ -33,41 +34,53 @@
 		di = (double)i;
 		v = X[ ix ];
 		sum += v;
 		d = (float)(di*(double)v) - sum;
 		S += (float)(1.0/(di*(di-1.0))) * d * d;
```
```suggestion
		d = (float)( di * (double) v ) - sum;
		S += (float)( 1.0 / ( di * ( di-1.0 ) ) ) * d * d;
```

aayush0325 replied:
we don't need this change here

Vinit-Pandit commented on lib/node_modules/@stdlib/stats/base/svarianceyc/benchmark/benchmark.js:
```diff
@@ -40,12 +46,8 @@ var svarianceyc = require( './../lib/svarianceyc.js' );
 */
 function createBenchmark( len ) {
 	var x;
```
```suggestion
```

Vinit-Pandit commented on lib/node_modules/@stdlib/stats/base/svarianceyc/benchmark/benchmark.js:
```diff
@@ -40,12 +46,8 @@ var svarianceyc = require( './../lib/svarianceyc.js' );
 function createBenchmark( len ) {
 	var x;
-	var i;
 
```
```suggestion
```

Vinit-Pandit commented on lib/node_modules/@stdlib/stats/base/svarianceyc/examples/index.js:
```diff
@@ -18,18 +18,14 @@
-var Float32Array = require( '@stdlib/array/float32' );
+var discreteUniform = require( '@stdlib/random/array/discrete-uniform' );
 var svarianceyc = require( './../lib' );
 
 var x;
```
```suggestion
```

Vinit-Pandit committed:
fix: cleanups

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svarianceyc/src/main.c:
```diff
@@ -78,8 +78,8 @@ float API_SUFFIX(stdlib_strided_svarianceyc_ndarray)( const CBLAS_INT N, const f
 		sum += v;
-		d = (float)(di*(double)v) - sum;
-		S += (float)(1.0/(di*(di-1.0))) * d * d;
+		d = (float)( di * (double)v ) - sum;
+		S += (float)( 1.0 / ( di * ( di-1.0 ) ) ) * d * d;
```
we don't need this change, this can be reverted back
```suggestion
		d = (float)(di*(double)v) - sum;
		S += (float)(1.0/(di*(di-1.0))) * d * d;
```

Vinit-Pandit committed:
revert: spacing

Co-authored-by: Aayush Khanna <96649223+aayush0325@users.noreply.github.com>
Signed-off-by: Vinit Pandit <106718914+MeastroZI@users.noreply.github.com>

aayush0325 commented on lib/node_modules/@stdlib/stats/base/svarianceyc/README.md:
```diff
@@ -224,6 +209,125 @@ console.log( v );
+
+#### stdlib_strided_svarianceyc_ndarray( N, correction, \*X, strideX, offsetX )
+
+Computes the [variance][variance] of a single-precision floating-point strided array using a one-pass algorithm proposed by Youngs and Cramer and using alternative indexing semantics.
```
```suggestion
Computes the [variance][variance] of a single-precision floating-point strided array using a one-pass algorithm proposed by Youngs and Cramer and alternative indexing semantics.
```

Vinit-Pandit committed:
Update lib/node_modules/@stdlib/stats/base/svarianceyc/README.md

Co-authored-by: Aayush Khanna <96649223+aayush0325@users.noreply.github.com>
Signed-off-by: Vinit Pandit <106718914+MeastroZI@users.noreply.github.com>

aayush0325 reviewed (approved):
lgtm

MeastroZI commented on lib/node_modules/@stdlib/stats/base/svarianceyc/docs/repl.txt:
```diff
@@ -110,10 +107,8 @@
     ~4.3333
 
     See Also
     --------
```
```suggestion
    --------
    
```

Vinit-Pandit committed:
style: add trailing space in repl txt

Signed-off-by: Vinit Pandit <106718914+MeastroZI@users.noreply.github.com>

Vinit-Pandit commented on lib/node_modules/@stdlib/stats/base/svarianceyc/docs/repl.txt:
```diff
@@ -110,10 +107,9 @@
     See Also
     --------
-
+    
```
```suggestion

```

Vinit-Pandit committed:
style: removing space

Signed-off-by: Vinit Pandit <106718914+MeastroZI@users.noreply.github.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/svarianceyc/README.md:
```diff
@@ -224,6 +209,125 @@ console.log( v );
+
+#### stdlib_strided_svarianceyc( N, correction, \*X, strideX )
+
+Computes the [variance][variance] of a single-precision floating-point strided array `x` using a one-pass algorithm proposed by Youngs and Cramer.
```
```suggestion
Computes the [variance][variance] of a single-precision floating-point strided array using a one-pass algorithm proposed by Youngs and Cramer.
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/svarianceyc/src/addon.c:
```diff
@@ -36,10 +36,27 @@ static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV_FLOAT( env, correction, argv, 1 );
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 3 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 2 );
+	STDLIB_NAPI_ARGV_INT64( env, offesetX, argv, 4 );
+	STDLIB_NAPI_CREATE_DOUBLE( env, (double)stdlib_strided_svarianceyc_ndarray( N, correction, X, strideX, offesetX ), v );
```
```suggestion
	STDLIB_NAPI_ARGV_INT64( env, offsetX, argv, 4 );
	STDLIB_NAPI_CREATE_DOUBLE( env, (double)stdlib_strided_svarianceyc_ndarray( N, correction, X, strideX, offsetX ), v );
```

kgryte committed:
refactor: fix typo

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/svarianceyc/src/addon.c:
```diff
@@ -36,10 +36,27 @@ static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_CREATE_DOUBLE( env, (double)stdlib_strided_svarianceyc_ndarray( N, correction, X, strideX, offsetX ), v );
+	return v;
+}
+STDLIB_NAPI_MODULE_EXPORT_FCN_WITH_METHOD( addon, "ndarray", addon_method )
```
```suggestion

STDLIB_NAPI_MODULE_EXPORT_FCN_WITH_METHOD( addon, "ndarray", addon_method )
```

kgryte committed:
style: add empty line

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/svarianceyc/src/main.c:
```diff
@@ -33,41 +34,53 @@
+    if ( N <= 0 || n <= 0.0f ) {
+        return 0.0f / 0.0f; // NaN
+    }
+    if ( N ==1 || strideX == 0 ) {
```
```suggestion
    if ( N == 1 || strideX == 0 ) {
```

kgryte committed:
style: add missing space

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM after minor clean-up.
